File: crypto-service/src/main/java/com/intela/cryptoservice/models/CandleStick.java
package com.intela.cryptoservice.models;

import com.binance.api.client.domain.market.Candlestick;
import lombok.*;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class CandleStick {
    private String symbol;
    private Double open_price;
    private Double high_price;
    private Double low_price;
    private Double volume;
    private Double close_price;
}

File: predict-service/src/model/candlestick.py
# src/model/candlestick.py

from marshmallow import Schema, fields

class Candlestick(Schema):
    symbol = fields.Str(required=True)
    volume = fields.Float(required=True)       # Adjusted to match numeric type
    open_price = fields.Float(required=True)   # Adjusted to match numeric type
    high_price = fields.Float(required=True)   # Adjusted to match numeric type
    low_price = fields.Float(required=True)    # Adjusted to match numeric type
    close_price = fields.Float(required=True)  # Adjusted to match numeric type

